mono/metadata/locales.c

```diff
@@ -530,22 +532,44 @@ get_current_locale_name (void)
 	return ret;
 }
 
-MonoString*
-ves_icall_System_Globalization_CultureInfo_get_current_locale_name (void)
+static MonoStringHandle
+culture_info_get_current_locale_name (MonoError *error)
 {
-	gchar *locale;
-	MonoString* ret;
+	MonoStringHandle result = MONO_HANDLE_NEW(MonoString, NULL);
```

Comment:
We need one here because `mono_string_new_handle` allocates a handle in the caller's frame for its return value.

I don't think new frames are going to be rare: any time managed code needs to take a reference to a managed object it will need a frame:
  1. If it calls any other function that returns a handle it will need (like above).
  2. If we are given a handle to some managed object but we need to work with one of its fields, we'll need a handle (and thus a new frame) to hold the reference.